### PR TITLE
ENH: allow NonLinLoc to make use of network code and location code

### PR DESCRIPTION
**Pull request by luca-s**
Currently NonLinLoc is not able to distinguish between multiple network
codes or location codes that use the same station code. This is due to
the fact that the station names are mapped to NonLinLoc configuration
using only the station code.

To get rid of this limitation, while keeping backward compatibility,
we introduce a new parameter (stationNameFormat) that allows to specify
how a station name is built: e.g. NETCODE_STACODE_LOCCODE

NETCODE, STACODE, LOCCODE are just placeholders and their value will be replaced
with the actual codes

As long as the user generates the grid file accordingly (e.g.
tablePath.P.NETCODE_STACODE_LOCCODE.time.*), this solves the problem

### CODE REVIEW

**Comment by luca-s** [comment]
@sheimers this should fix the limitation we discussed Wednesday

**Comment by luca-s** [comment]
@gempa-jabe I have used the code in this PR for some time now and it works fine. I would just like you to review the configuration option that I created. I feel like It is not super clear.

File: plugins/locator/nll/nll_locator.cpp [hunk 1]
```diff
@@ -426,6 +444,9 @@ bool NLLocator::init(const Config::Config &config) {
 		try { prof.tablePath = env->absolutePath(config.getString(prefix + "tablePath")); }
 		catch ( ... ) {}
 
+		try { prof.stationNameFormat = env->absolutePath(config.getString(prefix + "stationNameFormat")); }
```
Comment (gempa-jabe) [inline plugins/locator/nll/nll_locator.cpp]: Why `absolutePath`? Is that is filename?

File: plugins/locator/nll/nll_locator.cpp [hunk 2]
```diff
@@ -297,6 +298,23 @@ double normalizeLon(double lon) {
 
 
 
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+std::string stationName(Seiscomp::DataModel::Pick *pick, const std::string& rule)
+{
+  static const std::regex reNet("NETCODE", std::regex::optimize);
```
Comment (gempa-jabe) [inline plugins/locator/nll/nll_locator.cpp]: Actually we have `Utils::replace` which can be used to replace e.g.: "@NET@.@STA@.@LOC@". The prefix and postfix can be set to anything you want.

File: plugins/locator/nll/nll_locator.cpp (hunk 1 above)
Comment (luca-s) [reply plugins/locator/nll/nll_locator.cpp]: copy/paste  hits again. I will fix this thanks

File: plugins/locator/nll/nll_locator.cpp (hunk 2 above)
Comment (luca-s) [reply plugins/locator/nll/nll_locator.cpp]: Good to know. I'll try using `Utils::replace` then

**Comment by luca-s** [comment]
@gempa-jabe I updated the code. Please have a look, thanks.

**Comment by luca-s** [comment]
Now that Seiscomp 5 is out this PR will go into the next Seiscomp release. However I believe we could backport the feature to seiscomp 5 branch too, since it doesn't break backward compatibility

**Comment by luca-s** [comment]
@gempa-jabe I am going to backport this PR to v5 branch

**Comment by gempa-jabe** [comment]
You don't need to backport it. Just merge it into v5 if you want. Since it is an addon, it does not affect the SC API and it doesn't break the NLL configuration.

**Pull request merged**